Durf1985/microservices
# DO-170 Deployment minikube/GKE web interface

## In the process done:

* Launching reddit app in GKE with using Terraform

## How to setup the project:

* In GKE connect your cluster with your localhost 
```bash 
gcloud container clusters get-credentials your-gke-cluster-name --region your-region --project your-project
```

* In directory  `/your_repo_path/docker/docker-monolith/infra/terraform/kubernetes_cluster` execute `terraform init` and after `terraform apply`
* In directory `/your_repo_path/kubernetes/reddit` execute

```bash
kubectl apply -f dev-namespace # create namespace
kubectl apply -f ./ -n dev # deploy app in dev namespace
kubectl apply -f dashboard/ # create namespace  kubernetes-dashboard deploy dashboard in this namespace
kubectl apply -f s_account/  # create service account and binding cluster role
```

## How to check the project:

* execute `k get svc -n dev ui`,  `kubectl get nodes -o wide`
* open your browser  <http://node-ip:NodePort>
* for using dashboard check this documentation <https://kubernetes.io/docs/tasks/access-application-cluster/web-ui-dashboard/>

## PR checklist

The label with the homework number, in label description - lecture and topic of the homework
